Sources/Datadog/RUM/RUMMonitor/Scopes/RUMResourceScope.swift

```diff
@@ -208,6 +208,10 @@ internal class RUMResourceScope: RUMScope {
 
     private func sendErrorEvent(on command: RUMStopResourceWithErrorCommand) -> Bool {
         attributes.merge(rumCommandAttributes: command.attributes)
+        let sourceType = (attributes.removeValue(forKey: RUMAttribute.internalErrorSourceType) as? String)
+            .flatMap {
+                return RUMErrorEvent.Error.SourceType(rawValue: $0)
+            } ?? .ios
```

Comment:
We're missing unit tests for this logic. However, that would require similar complex test to the one added in `RUMViewScopeTests`. Wouldn't it be be better to resolve this attribute in [`RUMStopResourceWithErrorCommand`](https://github.com/DataDog/dd-sdk-ios/blob/3f98760beffeffc9f261a5dc9344a3d60ea6fb6b/Sources/Datadog/RUM/RUMMonitor/RUMCommand.swift#L168) and [`RUMAddCurrentViewErrorCommand`](https://github.com/DataDog/dd-sdk-ios/blob/3f98760beffeffc9f261a5dc9344a3d60ea6fb6b/Sources/Datadog/RUM/RUMMonitor/RUMCommand.swift#L60)? 

Note that other properties describing resource error and view error are transported as `RUMCommand` fields, produced upstream and only read downstream. If adding `sourceType` to `RUMStopResourceWithErrorCommand` and `RUMAddCurrentViewErrorCommand` we can easily test all cases in `RUMCommandTests`. If further making RUM scope tests fuzzy on `sourceType: .mockRandom()` that would be minimal effort for max safety. WDYT @buranmert @maxep ?

Reply:
done ✅